ENT-9380: Added doc for policy function filereadable (#2951)
cfengine/documentation: reference/functions/isreadable.markdown

---
layout: default
title: isreadable
published: true
tags: [reference, file functions, functions, files]
---

[%CFEngine_function_prototype(path, timeout)%]

> **Description:** Check if file is readable.

Reviewer: ```suggestion
**Description:** Check if a file is readable.
```

> This function checks if file specified by the `path` parameter is readable by

Reviewer: ```suggestion
This function checks if the file specified by the `path` parameter is readable by
```

> trying to read 1 Byte from it. If file is not ready for I/O, and the `timeout`

Reviewer: ```suggestion
trying to read 1 Byte from it. If the file is not ready for I/O, and the `timeout`
```

> parameter is not set to 0, the read operation will block for at most N number of
> seconds specified by the `timeout` parameter.

Reviewer: ```suggestion
parameter is not set to 0, the read operation will block for at most `timeout` number of seconds.
```

> The `timeout` parameter is optional and defaults to 3 seconds if it is not

Reviewer: ```suggestion
The `timeout` parameter is optional and defaults to 3 seconds, if it is not
```

specified. If the `timeout` parameter is set to 0, the read operation may block
indefinetly.

The function evaluates to false if the time limit expires or the read operation

> failes for any other reason (e.g. permission denied).

Reviewer: ```suggestion
fails for any other reason (e.g. permission denied).
```

**Notes:**

When the `timeout` parameter is set to 0, the agent will try to open and read
the file in the main thread, which can cause the agent to block indefinetly.

When the `timeout` parameter is not 0, the agent will spawn a separate thread
that will try to open and read the file. The agent will wait for at most N

> number of seconds for the spawned thread to finnish. If the thread does not

Reviewer: ```suggestion
number of seconds for the spawned thread to finish. If the thread does not
```

> finnish in time, the agent will cancel the thread execution, and concider the

Reviewer: ```suggestion
finish in time, the agent will cancel the thread execution, and consider the
```

Reviewer 2: got something against the fins? What if  it was sweedish in time?

file unreadable.

> 

Reviewer: ```suggestion

If the file is of size 0, the function will return true, if it successfully reads 0 bytes (reaches end-of-file).

```
